examples/react-client/prettier-chat/src/components/SettingsSheet.tsx

```diff
@@ -0,0 +1,41 @@
+import { FC, PropsWithChildren } from "react";
+import {
+  SheetTrigger,
+  SheetContent,
+  SheetHeader,
+  SheetTitle,
+  SheetDescription,
+  Sheet,
+} from "./ui/sheet";
+import { Label } from "./ui/label";
+import { CameraSettings, MicrophoneSettings } from "./DeviceSettings";
+
+export const SettingsSheet: FC<PropsWithChildren> = ({ children }) => {
+  return (
+    <Sheet>
+      <SheetTrigger>{children}</SheetTrigger>
+      <SheetContent>
+        <SheetHeader>
+          <SheetTitle>Settings</SheetTitle>
+          <SheetDescription>
+            Adjust your camera, microphone, and other settings.
+          </SheetDescription>
+        </SheetHeader>
+
+        <section className="space-y-4 mt-4">
+          <div className="space-y-2">
+            <Label>Camera settings</Label>
+
+            <CameraSettings />
```

Comment:
If the user clicks on `Camera settings` if access to camera is blocked the whole site will crash. Mic also has this problem
![image](https://github.com/user-attachments/assets/73b311b2-7afd-4ae0-befa-ee8405baf825)

Reply:
d